soh/src/overlays/actors/ovl_Bg_Toki_Swd/z_bg_toki_swd.c

```diff
@@ -122,7 +123,9 @@ void func_808BAF40(BgTokiSwd* this, PlayState* play) {
     if (!LINK_IS_ADULT || (Flags_GetEventChkInf(EVENTCHKINF_LEARNED_PRELUDE_OF_LIGHT) && !IS_RANDO) || IS_RANDO) {
         if (Actor_HasParent(&this->actor, play)) {
             if (!LINK_IS_ADULT) {
-                Item_Give(play, ITEM_SWORD_MASTER);
+                 if (!Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD)) {
+                    Item_Give(play, ITEM_SWORD_MASTER);
+                 }
```

Comment:
Vanilla code can enter this if block due to the `!IS_RANDO` above, and we need to avoid using `Randomizer_GetSettingValue` in a vanilla code flow. Should add another `!IS_RANDO` check here:

```suggestion
                if (!IS_RANDO || !Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD)) {
                    Item_Give(play, ITEM_SWORD_MASTER);
                }
```